app/src/main/java/com/yacgroup/yacguide/BaseNavigationActivity.kt

```diff
@@ -29,12 +30,15 @@ import androidx.drawerlayout.widget.DrawerLayout
 import com.google.android.material.navigation.NavigationView
 
 abstract class BaseNavigationActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
-    override fun onBackPressed() {
-        val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
-        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
-            drawerLayout.closeDrawer(GravityCompat.START)
-        } else {
-            super.onBackPressed()
+
+    private val onBackPressedCallback: OnBackPressedCallback = object: OnBackPressedCallback(true) {
```

Comment:
This does not need to (and should not) be a class member. Please define it within `onCreate()` where you use it:
```
onCreate() {
    ...
    onBackPressedDispatcher.addCallback(this, OnBackPressedCallback(true) {
        handleOnBackPressed() {
            ....
        }
    })
}